manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java

```diff
@@ -72,41 +76,61 @@ public Bridge createDependencies(Work work, Bridge bridge) {
         rhoasService.createTopicAndGrantAccessFor(resourceNamesProvider.getBridgeErrorTopicName(bridge.getId()),
                 RhoasTopicAccessType.CONSUMER_AND_PRODUCER);
 
-        // We don't need to wait for the Bridge to be READY to create the Error Handler.
-        createErrorHandlerProcessor(bridge);
+        // We don't need to wait for the Bridge to be READY to handle the Error Handler.
+        createOrUpdateOrDeleteErrorHandlerProcessor(bridge);
+
+        // Keep the Bridge in PREPARING until the Error Handler processor is READY.
+        // This ensures Users cannot update the Bridge Error Handler until it is available.
+        if (isErrorHandlerProcessorReady(bridge)) {
+            bridge.setDependencyStatus(ManagedResourceStatus.READY);
+        }
 
-        bridge.setDependencyStatus(ManagedResourceStatus.READY);
         return persist(bridge);
     }
 
+    private boolean isErrorHandlerProcessorReady(Bridge bridge) {
+        ListResult<Processor> hiddenProcessors = getHiddenProcessors(bridge);
+        return hiddenProcessors.getItems().stream().allMatch(ManagedResource::isActionable);
+    }
+
     /**
-     * Creates error handler processor if required
+     * Creates, updates or deletes Error Handler processor as required
      *
      * @param bridge input bridge
-     * @return true if the work can proceed (either the error handler processor
-     *         is not required or it's created and ready), false otherwise.
      */
-    private void createErrorHandlerProcessor(Bridge bridge) {
-        // If an ErrorHandler is not needed, consider it ready
+    private void createOrUpdateOrDeleteErrorHandlerProcessor(Bridge bridge) {
+        ListResult<Processor> hiddenProcessors = getHiddenProcessors(bridge);
+
+        // If an ErrorHandler is not defined, consider it ready and delete any lingering instances
         Action errorHandlerAction = bridge.getDefinition().getErrorHandler();
         boolean errorHandlerProcessorIsNotRequired = Objects.isNull(errorHandlerAction);
         if (errorHandlerProcessorIsNotRequired) {
+            deleteErrorHandlingProcessor(bridge);
             return;
         }
 
-        String bridgeId = bridge.getId();
-        String customerId = bridge.getCustomerId();
-        ListResult<Processor> processors = processorService.getHiddenProcessors(bridgeId, customerId);
-
+        // If an Error Handler processor exists assume it is to be updated otherwise create it!
         // This assumes we can only have one ErrorHandler Processor per Bridge
-        if (processors.getTotal() > 0) {
-            return;
+        if (hiddenProcessors.getTotal() > 0) {
```

Comment:
@manstis We could extract this into a `getErrorHandler` method which returns the `Optional` as we have similar logic in the delete flow. Or see my previous comment about providing a `getErrorHandler` method on the `ProcessorService` which feels perhaps more appropriate.

Reply:
Done. It does avoid a duplication of code in `BridgeWorker`.